src/components/Navrail/Navrail.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Search,
+  LayoutGrid,
+  CircleDollarSign,
+  Wallet,
+  Zap,
+  Gem,
+  Settings2,
+  CircleHelp,
+  Gift,
+} from 'lucide-react';
+import React, { FC } from 'react';
+import {
+  Sidebar,
+  SidebarContent,
+  SidebarFooter,
+  SidebarGroup,
+  SidebarGroupContent,
+  SidebarGroupLabel,
+  SidebarHeader,
+  SidebarMenu,
+  SidebarMenuButton,
+  SidebarMenuItem,
+  SidebarSeparator,
+  SidebarTrigger,
+  useSidebar,
+} from '@/components/ui/sidebar';
+
+import styles from './Navrail.module.scss';
+
+const menuOne = [
+  {
+    title: 'Search',
+    url: '#',
+    icon: Search,
+  },
+  {
+    title: 'Overview',
+    url: '#',
+    icon: LayoutGrid,
+  },
+  {
+    title: 'Expenses',
+    url: '#',
+    icon: CircleDollarSign,
+  },
+  {
+    title: 'Accounts',
+    url: '#',
+    icon: Wallet,
+  },
+  {
+    title: 'Insights',
+    url: '#',
+    icon: Zap,
+  },
+];
+
+const menuTwo = [
+  {
+    title: 'Settings',
+    url: '#',
+    icon: Settings2,
+  },
+  {
+    title: 'Chat for help',
+    url: '#',
+    icon: CircleHelp,
+  },
+  {
+    title: 'Refer and earn',
+    url: '#',
+    icon: Gift,
+  },
+];
+
+// Helper function to render menu items
+const renderSidebarMenuItems = (
+  items: { title: string; url: string; icon: React.FC }[]
+) => {
+  return items.map((item) => (
+    <SidebarMenuItem key={item.title}>
+      <SidebarMenuButton asChild>
+        <a href={item.url}>
+          <item.icon size={18} /> {/* Set icon size */}
```

Comment:
passing size isn't doing anything here, the default size is 16 for `<item.icon />` so you can just do
```suggestion
          <item.icon />
```